src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -1332,11 +1334,14 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 		}
 
 		// Settings
-		const tabActionsVisibility = isTabSticky && options.pinnedTabSizing === 'compact' ? 'off' /* treat sticky compact tabs as tabCloseButton: 'off' */ : options.tabCloseButton;
+		const tabActionPosition = hasUnpinAction ? options.tabUnpinButton : options.tabCloseButton;
+		const tabActionsVisibility = isTabSticky && options.pinnedTabSizing === 'compact' ? 'off' : tabActionPosition; /* treat sticky compact tabs as tabCloseButton/tabUnpinButton: 'off' */
```

Comment:
This we need to devide up. If `!hasUnpinAction` we need to check if close action should be visible. If not then toggle `tab-actions-off` (probably should be renamed).
Then check the location and set `tab-actions-right` / `tab-actions-left`. This probably should only be done if an action is visible.